kafka-rest/src/main/java/io/confluent/kafkarest/response/StreamingResponse.java

```diff
@@ -218,11 +225,15 @@ private void closeAll(AsyncResponseQueue responseQueue) {
     responseQueue.close();
   }
 
-  private ResultOrError handleNext(T result, @Nullable Throwable error) {
+  private ResultOrError handleNext(
+      T result, @Nullable Throwable error, ProduceRecordErrorCounter produceRecordErrorCounter) {
     if (error == null) {
+      produceRecordErrorCounter.incrementErrorCount(HTTP_SUCCESS_CODE);
       return ResultOrError.result(result);
     } else {
       log.debug("Error processing streaming operation.", error);
+      int errorCode = EXCEPTION_MAPPER.toErrorResponse(error.getCause()).getErrorCode();
```

Comment:
do we already check if `error.getCause()` is not null or the whole `EXCEPTION_MAPPER.toErrorResponse(error.getCause()).getErrorCode()` could throw exception?

Reply:
```suggestion
      if (error.getCause() == null) {
        throw new IllegalArgumentException("Error cause is null", error);
      }
      int errorCode = EXCEPTION_MAPPER.toErrorResponse(error.getCause()).getErrorCode();
      produceRecordErrorCounter.incrementErrorCount(errorCode);
```

Thanks for calling this! Is this change above good?